src/foam/nanos/controller/ApplicationController.js

```diff
@@ -483,24 +484,32 @@ foam.CLASS({
         } else {
           // If they don't, then we fetch the most appropriate theme based on
           // a few different parameters.
+          var predicate = this.TRUE;
           var predicates = [];
+          var themeDAO = this.client.themeDAO;
+          var themeDomainDAO = this.client.themeDomainDAO;
 
+          // legacy support
           if ( this.webApp ) {
             predicates.push(this.EQ(this.Theme.APP_NAME, this.webApp));
+            try {
+              this.theme = await themeDAO.find(predicate);
+              if ( ! lastTheme || lastTheme.id != this.theme.id ) this.useCustomElements();
+              return;
+            } catch (err) {
+              console.warning(err);
+              // continue
```

Comment:
So with this logic the application controller will consider `personalTheme` if none is found then use `webApp`. If the `personalTheme` is found then we use the `domainTheme`. Seems kind of odd since we're disregarding the personalTheme altogether.

I think the cascade from 1st being the one applied should be personalTheme -> webApp? (Do we need this if we have domain themes?) -> domainTheme

Reply:
webapp has been removed. 
personal theme is tested first and we fall through if it doesn't exist, just like it was before. 
A domain will only have one theme.